fish.go

```diff
@@ -163,14 +165,28 @@ func fishSubcommandHelper(binary string, command *Command, siblings []*Command)
 func fishFlagHelper(binary string, command *Command) string {
 	fishHelper := fmt.Sprintf("__fish_%s_no_subcommand", binary)
 	if len(command.Lineage()) > 1 {
-		fishHelper = fmt.Sprintf(
-			"__fish_seen_subcommand_from %s",
-			strings.Join(command.Names(), " "),
-		)
+		fishHelper = commandAncestry(command)
 	}
 	return fishHelper
 }
 
+func commandAncestry(command *Command) string {
+	var ancestry []string
+	ancestors := command.Lineage()
+	slices.Reverse(ancestors)
+	ancestors = ancestors[1:]
+	for _, ancestor := range ancestors {
+		ancestry = append(
+			ancestry,
+			fmt.Sprintf(
+				"__fish_seen_subcommand_from %s",
+				strings.Join(ancestor.Names(), " "),
+			),
+		)
+	}
```

Comment:
I think it is simpler to use a reverse loop.

```suggestion
	for i := len(ancestors) - 2; i >= 0; i-- {
		ancestry = append(
			ancestry,
			fmt.Sprintf(
				"__fish_seen_subcommand_from %s",
				strings.Join(ancestors[i].Names(), " "),
			),
		)
	}
```